server/integration/infinispan/src/main/java/org/jboss/as/clustering/infinispan/subsystem/InfinispanSubsystemXMLWriter.java

```diff
@@ -204,7 +207,69 @@ private static void writeScheduledThreadPoolElements(Element element, ScheduledT
         }
     }
 
-    private void processCacheConfiguration(XMLExtendedStreamWriter writer, ModelNode container, ModelNode configurations, String cacheType)
+   private void processCounterConfigurations(XMLExtendedStreamWriter writer, ModelNode container)
+         throws XMLStreamException {
+
+      if (container.hasDefined(ModelKeys.COUNTERS)) {
+         writer.writeStartElement(Element.COUNTERS.getLocalName());
+
+         //counters element and its attributes
+         ModelNode counterRoot = container.get(ModelKeys.COUNTERS);
+         this.writeOptional(writer, Attribute.RELIABILITY, counterRoot, ModelKeys.RELIABILITY);
+         this.writeOptional(writer, Attribute.NUM_OWNERS, counterRoot, ModelKeys.NUM_OWNERS);
+
+         //all counters configurations
+         ModelNode counterConfigurations = counterRoot.get(ModelKeys.COUNTERS_NAME);
+         processStrongCounterConfigurations(writer, counterConfigurations.get(ModelKeys.STRONG_COUNTER));
+         processWeakCounterConfigurations(writer, counterConfigurations.get(ModelKeys.WEAK_COUNTER));
+         writer.writeEndElement();
+      }
+   }
+
+   private void processWeakCounterConfigurations(XMLExtendedStreamWriter writer, ModelNode configurations) throws XMLStreamException {
+      if (configurations != null &&  configurations.isDefined()) {
+         for (Property e : configurations.asPropertyList()) {
+            processWeakCounterConfiguration(writer, e.getValue());
+         }
+      }
+   }
+
+   private void processStrongCounterConfigurations(XMLExtendedStreamWriter writer, ModelNode configurations) throws XMLStreamException {
+      if (configurations != null && configurations.isDefined()) {
+         for (Property e : configurations.asPropertyList()) {
+            processStrongCounterConfiguration(writer, e.getValue());
+         }
+      }
+   }
+
+   private void processWeakCounterConfiguration(XMLExtendedStreamWriter writer, ModelNode weakConfiguration) throws XMLStreamException {
+      writer.writeStartElement(Element.WEAK_COUNTER.getLocalName());
+      this.writeRequired(writer, Attribute.NAME, weakConfiguration, ModelKeys.NAME);
+      this.writeRequired(writer, Attribute.INITIAL_VALUE, weakConfiguration, ModelKeys.INITIAL_VALUE);
```

Comment:
the only required attribute is the name. the remaining attributes are optional.

Reply:
ok, thanks @pruivo